shell/src/main/java/alluxio/shell/command/LeaderCommand.java

```diff
@@ -44,16 +46,16 @@ public int getNumOfArgs() {
 
   @Override
   public int run(CommandLine cl) {
-    FileSystemMasterClient client = FileSystemContext.INSTANCE.acquireMasterClient();
-    try {
+    try (FileSystemMasterClient client = FileSystemContext.INSTANCE.acquireMasterClient()) {
```

Comment:
@Reidddddd I was not clear in the previous comment. In here, the "cleaner" way is to acquire the master client resource, not the master client. difference here is with master client resource, on its close method, the resource represented by this client will be returned to the pool automatically. In contrast, if you `acquireMasterClient()`, you need to explicitly return the resource.

Thus , the code should look like:
```java
 try (CloseableResource<FileSystemMasterClient> client = FileSystemContext.INSTANCE.acquireMasterClientResource()) {
  String hostName = client.get().getAddress().getHostName();
  if (hostName != null) {
    System.out.println(hostName);
  } else {
    System.out.println("Failed to get the leader master.");
  }
  // NOTE, you don't need to release the master client here
} catch (IOException e) {
  System.err.println("Failed to get the leader master due to interruption occured.");
}
```